File: JavaScript/Math-Array-String/61.js
// 2011. Final Value of Variable After Performing Operations


/**
 * @param {string[]} operations
 * @return {number}
 */
var finalValueAfterOperations = function(operations) {
    let X = 0; // Initialize X to 0
    for (let operation of operations) {
        if (operation === "++X" || operation === "X++") {
            X += 1;
        } else if (operation === "--X" || operation === "X--") {
            X -= 1;
        }
    }
    return X; // Return the final value of X
};


let operations = ["--X", "X++", "X++"];
console.log(finalValueAfterOperations(operations)); // Output: 1

File: Python/Math-Array-String/61.py
# 2011. Final Value of Variable After Performing Operations

from typing import List

class Solution:
    def finalValueAfterOperations(self, operations: List[str]) -> int:
        X = 0  # Initialize X to 0
        for operation in operations:
            if operation in ("++X", "X++"):
                X += 1
            elif operation in ("--X", "X--"):
                X -= 1
        return X
    

sol = Solution()
output = sol.finalValueAfterOperations(["--X", "X++", "X++"])
print(output) # # Output: 1